packages/ses/src/error/assert.js

```diff
@@ -401,6 +401,45 @@ const makeAssert = (optRaise = undefined, unredacted = false) => {
   const assertString = (specimen, optDetails) =>
     assertTypeof(specimen, 'string', optDetails);
 
+  /** @type {NotThrows} */
+  const notThrows = action => {
```

Comment:
This does not seem to add much over having the programmer write `try/catch`.  That would be understandable by more developers with no additional information. Have the catch block be `fatalCommitFailure` or `throw vat.inconsistentCommit`.  I like getting a consistent API for terminating (`vat.assert`). That should also support a version of assert that takes a promise and terminates the scope if the promise eventually break.

Reply:
> This does not seem to add much over having the programmer write try/catch.

I agree, just came to the same conclusion from your comments on https://github.com/Agoric/agoric-sdk/pull/3287 , and closed this PR. Thanks!

> That should also support a version of assert that takes a promise and terminates the scope if the promise eventually break.

The case where this seems to arise in https://github.com/Agoric/agoric-sdk/pull/3287 is not best dealt with that way. Rather, for such errors that occur on the zoe side in service of the zcf vat, it should be the zoe side that decides to terminate that zcf vat.

The case you raise still seems plausible, but let's wait until we actually encounter a good example.